PKHeX.Core.AutoMod/AutoMod/APILegality.cs

```diff
@@ -1837,7 +1837,13 @@ public static void FixVivillonRegion(this PKM pk)
         /// <summary>
         /// Wrapper function for GetLegalFromTemplate but with a Timeout
         /// </summary>
-        public static AsyncLegalizationResult GetLegalFromTemplateTimeout(this ITrainerInfo dest, PKM template, IBattleTemplate set, bool nativeOnly = false)
+        public static AsyncLegalizationResult GetLegalFromTemplateTimeout(this ITrainerInfo dest, PKM template, IBattleTemplate set, bool nativeOnly = false) =>
+            GetLegalFromTemplateTimeoutAsync(dest, template, set, nativeOnly)
+                .ConfigureAwait(false)
```

Comment:
**note:** again, thanks for the heads up, opened up both solutions together (PKHeX and AutoLegalityMod) for a debug and I realized that I was missing the `ConfigureAwait(false)`

Without it, the task was being executed on the main thread, and because Windows Forms uses the main thread as the UI thread, it was causing a deadlock.

Offloading the task to the thread pool with the `ConfigureAwait` solves the issue

Tested the auto-legality and the living dex and both are working now